smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/PublishEndToEndIT.java

```diff
@@ -0,0 +1,332 @@
+
+    protected static final String DIST_AGENTS_PATH = "/libs/sling/distribution/services/agents";
+    protected static final String PUBLISH_DIST_AGENT = "publish";
+    protected static final String QUEUES_PATH = DIST_AGENTS_PATH + "/" + PUBLISH_DIST_AGENT;
+
+    @ClassRule
+    public static CQAuthorPublishClassRule cqBaseClassRule = new CQAuthorPublishClassRule();
+
+    @Rule
+    public CQRule cqBaseRule = new CQRule(cqBaseClassRule.authorRule, cqBaseClassRule.publishRule);
+
+    @Rule
+    public Page root = new Page(cqBaseClassRule.authorRule);
+
+    static CQClient adminAuthor;
+    static CQClient adminPublish;
+    static CQClient anonymousPublish;
+
+    private static ReplicationClient rClient;
+
+    @BeforeClass
+    public static void beforeClass() throws ClientException, IOException {
+        adminAuthor = cqBaseClassRule.authorRule.getAdminClient(CQClient.class);
+        adminPublish = cqBaseClassRule.publishRule.getAdminClient(CQClient.class);
+        anonymousPublish = cqBaseClassRule.publishRule.getClient(CQClient.class, null, null);
+        rClient = adminAuthor.adaptTo(ReplicationClient.class);
+    }
+
+    @Before
+    public void before() throws TimeoutException, InterruptedException {
+        new Polling(this::checkContentDistributionAgentExists)
+        	.poll(TIMEOUT, 500);
+    }
+
+	/**
+     * Activates a page as admin, then deactivates it.
+     * Verifies:
+     * <ul>
+     *      <li>That the replication queue is empty</li>
+     *      <li>If check enabled verifies that deleted page gets removed from publish</li>
+     * </ul>
+     *
+     * @throws Exception if an error occurred
+     */
+    @Test
+    public void testActivateAndDeactivate() throws Exception {
+
+        // This is a workaround for correctly detecting assumption violations.
+        // Any AssumptionViolatedException throw by the the Callable will be
+        // swallowed by the Poller and wrapped in a TimeoutException. As a
+        // consequence, the test will be marked as failed instead of skipped.
+
+        try {
+            activateAndDeactivate();
+        } catch (AssumptionViolatedException e) {
+            throw e;
+        } catch (Exception e) {
+            new Polling(this::activateAndDeactivate).poll(TIMEOUT, 500);
+        }
+    }
+    
+    /**
+     * Activates a page as admin, then deletes it.
+     * Verifies:
+     * <ul>
+     *  <li>That the replication queue is empty</li>
+     *  <li>If check enabled verifies that deleted page gets removed from publish</li>
+     * </ul>
+     * @throws Exception if an error occurred
+     */
+    @Test
+    public void testActivateAndDelete() throws Exception {
+
+        // This is a workaround for correctly detecting assumption violations.
+        // Any AssumptionViolatedException throw by the the Callable will be
+        // swallowed by the Poller and wrapped in a TimeoutException. As a
```

Comment:
Thanks for catching this. I'll open an issue in sling

Reply:
But why not make the methods like `activateAndDelete` throw the AssumtionViolatedException?

Reply:
This was existing code so, I did not catch this :)
But I still don't understand your point. The point is that any exception would be wrapped as a TimeoutException in the `Polling.poll()` method if invoked directly.